source/config/Config.cpp

```diff
@@ -1562,6 +1563,12 @@ bool PlainConfig::PubSub::LoadFromJson(const Crt::JsonView &json)
         }
     }
 
+    jsonKey = JSON_PUB_SUB_PUBLISH_ON_CHANGE;
+    if (json.ValueExists(jsonKey))
+    {
+        publishOnChange = json.GetBool(jsonKey);
```

Comment:
add details regarding this in help message: https://github.com/awslabs/aws-iot-device-client/blob/main/source/config/Config.cpp#L2867-L2930

Reply:
Added a help message for the addition.

Reply:
As an aside, is there a README to get the container with the c-lint rules or what the ruleset is? It's feels a bit like guesswork to get it right.